core/node/fee_model/src/l1_gas_price/main_node_fetcher.rs

```diff
@@ -73,6 +99,29 @@ impl MainNodeFeeParamsFetcher {
 
 impl BatchFeeModelInputProvider for MainNodeFeeParamsFetcher {
     fn get_fee_model_params(&self) -> FeeParams {
-        *self.main_node_fee_params.read().unwrap()
+        let fee_params = *self.main_node_fee_params.read().unwrap();
+        let batch_fee_input = *self.main_node_batch_fee_input.read().unwrap();
+        if batch_fee_input.is_none() {
+            return fee_params;
+        }
+        let batch_fee_input = batch_fee_input.unwrap();
+        match fee_params {
+            FeeParams::V1(..) => FeeParams::V1(FeeParamsV1 {
+                config: FeeModelConfigV1 {
+                    minimal_l2_gas_price: batch_fee_input.fair_l2_gas_price(),
+                },
+                l1_gas_price: batch_fee_input.l1_gas_price(),
+            }),
+            FeeParams::V2(params) => {
+                let mut config = params.config();
+                config.minimal_l2_gas_price = batch_fee_input.fair_l2_gas_price();
```

Comment:
this is incorrect: `minimal_l2_gas_price` and `fair_l2_gas_price` are different. There is no easy way to deduct `minimal_l2_gas_price` from `batch_fee_input`